api/v1/services/TrackingService.js

```diff
@@ -0,0 +1,82 @@
+var CheckitError = require('checkit').Error;
+var _ = require('lodash');
+var _Promise = require('bluebird');
+
+var cache = require('../../cache').instance();
+var TrackingItem = require('../models/UniversalTrackingItem');
+var errors = require('../errors');
+var utils = require('../utils');
+
+const TrackingFooter = "UniversalTracking";
+
+module.exports.createTrackingEvent = function (attributes) {
+    var trackingItem = TrackingItem.forge(attributes);
+
+    return trackingItem
+        .validate()
+        .catch(CheckitError, utils.errors.handleValidationError)
+        .then(function (validated) {
+            return TrackingItem.findByName(attributes.name);
+        })
+        .then(function (result) {
+            if (!_.isNull(result)) {
+                var message = "This event is already being tracked";
+                var source = "name";
+                throw new errors.InvalidParameterError(message, source);
+            }
+
+            return cache.getAsync('trackedEvent')
+        })
+        .then(function (result) {
+            if(!_.isNil(result)) {
+                return cache.ttlAsync('trackedEvent')
+                    .then(function (ttl) {
+                        var message = "An event is already being tracked at this time, seconds left: " + ttl;
+                        var source = trackingItem.get('name');
+                        return _Promise.reject(new errors.InvalidTrackingStateError(message, source));
+                    });
+            }
+
+            return cache.multi()
+                .set('trackedEvent', trackingItem.get('name'))
+                .expire('trackedEvent', trackingItem.get('duration'))
+                .execAsync()
+                .then(function () {
+                    return trackingItem.save();
+                });;
+        })
+};
+
+module.exports.addEventParticipant = function(participantId) {
+    return cache.getAsync('trackedEvent')
+        .then(function (currentEvent) {
+            if(_.isNil(currentEvent)) {
+                var message = "No event is currently being tracked";
+                var source = "EventTracking";
+                throw new errors.InvalidTrackingStateError(message, source);
+            }
+
+            return cache.getAsync(participantId + TrackingFooter)
+                .then(function (result) {
+                    if(!_.isNil(result)) {
+                        var message = 'This attendee has already participated in this event!';
+                        var source = participantId;
+                        return _Promise.reject(new errors.InvalidParameterError(message, source));
```

Comment:
This looks like it would give us an error response, right? What would the error code be?

Also, it would be helpful to send the name of the event as well in the error.